modules/cost/scheduled-action/main.bicep

```diff
@@ -0,0 +1,128 @@
+// https://learn.microsoft.com/rest/api/cost-management/scheduled-actions/create-or-update
+
+/**
+ * Parameters
+ */
+
+// TODO: targetScope = 'subscription'
+
+@description('Required. The name of the scheduled action.')
+param scheduledActionName string
+
+@description('Optional. Indicates the kind of scheduled action. Allowed: Email, InsightAlert. Default: InsightAlert.')
+@allowed([
+  'Email'
+  'InsightAlert'
+])
+param kind string = 'InsightAlert'
+
+@description('Optional. Specifies which built-in view to use. This is a shortcut for the full view ID. Allowed: AccumulatedCosts, DailyCosts, InvoiceDetails. Ignored if kind is "InsightAlert".')
+@allowed([
+  ''
+  'AccumulatedCosts'
+  'DailyCosts'
+  'InvoiceDetails'
+])
+param builtInView string = ''
+
+@description('Required if kind is "Email" and builtInView is not set. The resource ID of the view to which the scheduled action will send. The view must either be private (tenant level) or owned by the same scope as the scheduled action. Ignored if kind is "InsightAlert" or if builtInView is set.')
+param viewId string = ''
+var internalViewId = builtInView == null ? viewId : '${subscription().id}/providers/Microsoft.CostManagement/views/ms:${builtInView}'
```

Comment:
I've seen that recommendation before. I don't love that because it forces things that should be managed together to be separate. What's the reasoning behind the separation?

Reply:
Yeah, i share your opinion. However through contributing here and in the Azure QuickStart Templates - i've fallen in line 😅

Reasoning is consistency and maintainability.